### PR TITLE
Move `getTensorBaseAddress` to `IrBuilder`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
So that it can be used in other files

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/ir/builder.h [hunk 1]
```diff
@@ -100,6 +100,10 @@ class IrBuilder {
   // Get tensor metadata
   static Val* metadataExpr(TensorView* tv);
 
+  // Get tensor base address, for gmem tensor, it is something like
+  // `T1.data`. For smem tensor, it is something like `toSmem(T1)`.
+  static Val* tensorBaseAddressExpr(TensorView* tv);
```
Comment (wujingyue) [inline csrc/ir/builder.h:105]: Naming nit: `tensor` in the name seems a bit redundant. 

```suggestion
  static Val* baseAddressExpr(TensorView* tv);
```

**Review by wujingyue — APPROVED** [review]

File: csrc/ir/builder.h (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/ir/builder.h]: Makes sense, changed

**Comment by zasdfgbnm** [comment]
!build

**Review by drzejan2 — COMMENTED** [review]
Changes look good, thank you for preparing them.

**Pull request merged**